### PR TITLE
Upgrade S6-overlay from v1 to v3

### LINKED ISSUE #230: Upgrade S6 overlay to v3

### PR DESCRIPTION
**Pull request by mikedep333**
from 1.22.1.0 to 3.1.2.0

fixes: #230

### CODE REVIEW

File: .github/workflows/pr_build_images.yaml [hunk 1]
```diff
@@ -27,7 +27,12 @@ jobs:
       - uses: actions/checkout@v2
       - name: Build the images
         run: |
-          wget https://github.com/just-containers/s6-overlay/releases/download/v1.22.1.0/s6-overlay-amd64.tar.gz
+          wget https://github.com/just-containers/s6-overlay/releases/download/v3.1.1.2/s6-overlay-x86_64.tar.xz
+          wget https://github.com/just-containers/s6-overlay/releases/download/v3.1.1.2/s6-overlay-noarch.tar.xz
+          wget https://github.com/just-containers/s6-overlay/releases/download/v3.1.1.2/s6-overlay-symlinks-arch.tar.xz
+          wget https://github.com/just-containers/s6-overlay/releases/download/v3.1.1.2/s6-overlay-symlinks-noarch.tar.xz
+          unxz s6-overlay*.tar.xz
+          gzip s6-overlay*.tar
```
Comment (mdellweg) [inline .github/workflows/pr_build_images.yaml:35]: Does tar not support xz in the container?

Comment (mikedep333) [reply .github/workflows/pr_build_images.yaml:35]: No, and I'd rather save container size than install it.

File: pulp_ci_centos/Containerfile [hunk 2]
```diff
@@ -2,10 +2,15 @@ FROM quay.io/centos/centos:stream8
...
+COPY s6-overlay-noarch.tar.gz /tmp
+COPY s6-overlay-symlinks-noarch.tar.gz /tmp
+COPY s6-overlay-symlinks-arch.tar.gz /tmp
 
-RUN tar xzf /tmp/s6-overlay-amd64.tar.gz -C / --exclude="./bin" && \
-    tar xzf /tmp/s6-overlay-amd64.tar.gz -C /usr ./bin
+RUN tar xzf /tmp/s6-overlay-x86_64.tar.gz -C /
+RUN tar xvf /tmp/s6-overlay-noarch.tar.gz -C /
+RUN tar xvf /tmp/s6-overlay-symlinks-noarch.tar.gz -C /
+RUN tar xvf /tmp/s6-overlay-symlinks-arch.tar.gz -C /
```
Comment (mdellweg) [inline pulp_ci_centos/Containerfile]: Would we save more space by removing those files again?

Comment (mikedep333) [reply pulp_ci_centos/Containerfile]: I think it depends on whether layers are in use or not. Either way, I'll try removing them immediately afterwards.

File: assets/services/certs/up [hunk 3]
```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/execlineb -P
+define PREFFIX "[oneshot] certs:"
 
+foreground {
+  if { s6-echo "${PREFFIX} ${C031}begin generating certs${C000}" }
```
Comment (mdellweg) [inline assets/services/certs/up]: What is the reason to run this as `if` ?

File: assets/test.sh [hunk 4]
```diff
@@ -30,7 +30,7 @@ docker run --detach \
            --device /dev/fuse \
            "$image"
 sleep 10
-for _ in $(seq 20)
+for _ in $(seq 40)
```
Comment (mdellweg) [inline assets/test.sh]: Can you explain this change? Is the new s6 startup so much worse?

If the container took 10+ minutes to start up, i would consider it broken.

Comment (mikedep333) [reply assets/test.sh]: This is leftover from my earlier attempts to fix the container.

I changed it back.

**Review by dkliban — COMMENTED** [review]
This all looks great! Can you fix the spelling of PREFFIX to PREFIX?

File: assets/services/certs/up (hunk 3 above)
Comment (mikedep333) [reply assets/services/certs/up]: That's how it was done in postgres-prepare/up.

I removed it from this script, it still works.

File: assets/test.sh (hunk 4 above)
Comment (mikedep333) [reply assets/test.sh]: I changed it to 30 to avoid a timeout on one of the 4 CI tests.

This causes it to sleep for a total of 100 seconds.

**Review by dkliban — APPROVED** [review]

**Pull request merged**